src/core/sequence_batch_scheduler.cc

```diff
@@ -845,12 +845,13 @@ SequenceBatchScheduler::SequenceBatch::SchedulerThread(
     }
 
     if ((payloads != nullptr) && !payloads->empty()) {
-      auto OnCompleteQueuedPayloads = [payloads](Status status) {
+      auto OnCompleteQueuedPayloads = [payloads](const Status& rstatus) {
```

Comment:
I have to admit, this is something new to me. What is going on with `[ ]` around `payloads`? Is this a type of `shared_ptr` dereference?

Reply:
Its a lambda (anonymous function). The square brackets capture variables from the local env so that they are available when the function executes. It is up to dev to make sure that the captured variables are still valid when the lambda executes. payloads is a shared_ptr in this case which ensures that it remains live for the lambda to use.